test/test_art.py

```diff
@@ -416,6 +506,30 @@ def test_google_art_returns_no_result_with_malformed_response(self):
             next(self.source.get(album, self.settings, []))
 
 
+class CoverArtArchiveTest(UseThePlugin, CAAHelper):
+
+    def setUp(self):
+        super(CoverArtArchiveTest, self).setUp()
+        self.source = fetchart.CoverArtArchive(logger, self.plugin.config)
+        self.settings = Settings(maxwidth=0)
+
+    @responses.activate
+    def run(self, *args, **kwargs):
+        super(CoverArtArchiveTest, self).run(*args, **kwargs)
+
+    def test_caa_finds_image(self):
+        album = _common.Bag(mb_albumid=self.MBID_RELASE,
+                            mb_releasegroupid=self.MBID_GROUP)
+        self.mock_caa_response(self.RELEASE_URL, self.RESPONSE_RELEASE)
+        self.mock_caa_response(self.GROUP_URL, self.RESPONSE_GROUP)
+        candidate = next(self.source.get(album, self.settings, []))
+        self.assertIsNotNone(candidate)
+        self.assertEqual(candidate.url,
+                         'http://coverartarchive.org/release/rid/12345.gif')
+        self.assertEqual(len(responses.calls), 1)
+        self.assertEqual(responses.calls[0].request.url, self.RELEASE_URL)
```

Comment:
I guess this test would be more meaningful if it checked that `list(self.source.get(album, self.settings, []))` contains both images, which is what this PR is trying to achieve in the first place. the GIF is what the old implementation would have returned, too.

Reply:
This should actually have been checking for three images. Two for the release and one for the release group. Fixed.